### PR TITLE
Using Socket instead of HTTP connection

### PR DESCRIPTION
**Pull request by stevyhacker**
- Dependencies have been updated and tested on the sample project

- testNetworkRequest refactored to use a Socket connection and ping Google DNS for faster internet connectivity test than the current implementation via an HTTPUrlConnection

### CODE REVIEW

File: connectionbuddy/src/main/java/com/zplesac/connectionbuddy/ConnectionBuddy.java [hunk 1]
```diff
@@ -339,29 +336,16 @@ private void testNetworkRequest(@NonNull final NetworkRequestCheckListener liste
...
-                            public void run() {
-                                listener.onNoResponse();
-                            }
-                        });
-                    }
+                    Socket socket = new Socket();
+                    SocketAddress socketAddress = new InetSocketAddress(NETWORK_CHECK_URL_GOOGLE_DNS, DNS_PORT);
+                    socket.connect(socketAddress, CONNECTION_TIMEOUT);
+                    socket.close();
+                    callbackExecutor.execute(new Runnable() {
```
Comment (zplesac) [inline connectionbuddy/src/main/java/com/zplesac/connectionbuddy/ConnectionBuddy.java:343]: What happens here if the API call is not successfully executed? Does Socket always returns IOException in this case?

Could you maybe provide metrics for this? I'm interested in profiling how much is this implementation faster than the previous one.

Comment (stevyhacker) [reply connectionbuddy/src/main/java/com/zplesac/connectionbuddy/ConnectionBuddy.java:343]: Here is a comparison from Network Profiler: 

 first the old `HttpUrlConnection` implementation:

Screenshot_2020-10-14 11 36 16_LHnXeA

and then the `Socket` implementation:

Screenshot_2020-10-14 11 35 53_1N0pRw

The size of the payload is noticeably smaller in the new implementation and the speed is also visibly better.

As for the situation where the API call is not successfully executed, it does return IOException. It's available on all Android API levels as it's a standard Java class.

Also, this implementation is quite popular on StackOverflow :) https://stackoverflow.com/questions/1560788/how-to-check-internet-access-on-android-inetaddress-never-times-out

**Review by frankot-inf — APPROVED** [review]
Looks good to me, nice work. 👍

**Review by JonatanPlesko — APPROVED** [review]

File: connectionbuddy/src/main/java/com/zplesac/connectionbuddy/ConnectionBuddy.java (hunk 1 above)
Comment (zplesac) [reply connectionbuddy/src/main/java/com/zplesac/connectionbuddy/ConnectionBuddy.java:343]: Wow, nice, I really dig this.
Could you also add to the method documentation that the network request is executed on a separate, non-UI thread?

Comment (stevyhacker) [reply connectionbuddy/src/main/java/com/zplesac/connectionbuddy/ConnectionBuddy.java:343]: Sure, I have updated the method docs.

**Review by zplesac — APPROVED** [review]

**Pull request merged**